x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java

```diff
@@ -200,9 +204,9 @@ protected void nodeOperation(AllocatedPersistentTask task, @Nullable DataFrameTr
                 final long lastCheckpoint = stateHolder.get().getCheckpoint();
 
                 if (lastCheckpoint == 0) {
-                    logger.trace("[{}] No checkpoint found, starting the task", transformId);
-                    startTask(buildTask, indexerBuilder, lastCheckpoint, startTaskListener);
```

Comment:
this is the main fix: we started the task without loading the next checkpoint when last checkpoint is 0,